### PR TITLE
[Feature] add advanced place method to automata core when compass peripheral is equipped

### LINKED ISSUE #599: Impossible to craft compass turtle
### Describe

Hi ! I see "compass turtle" in JEI but i'm unable to craft it with a turtle and a compass... Is it a bug or something else ? Thanks for your anwser !

Magrets

### Steps to reproduce

Take a turtle, take a compass et try to craft both together or with " turtle.equipLeft()" it says "not a valid upgrade"

### Multiplayer?

No

### Version

1.20.1-0.7.38r

### Minecraft, Forge and maybe other related mods versions

Forge  47.2.21| Minecraft 1.20.1

### Screenshots or Videos

_No response_

### Crashlog/log

_No response_

### PR DESCRIPTION
**Pull request by zyxkad**
**PLEASE READ THE [GUIDELINES](https://github.com/SirEndii/AdvancedPeripherals/blob/dev/1.20.1/CONTRIBUTING.md) BEFORE MAKING A CONTRIBUTION**

* **Please check if the PR fulfills these requirements**
- [x] The commit message are well described
- [x] Docs have been added / updated (for features or maybe bugs which were noted). If not, please update the needed documentation [here](https://github.com/SirEndii/Advanced-Peripherals-Documentation/pulls). This is not mandatory
- [x] All changes have fully been tested

* **What kind of change does this PR introduce?** (Bug fix, feature, ...)
Feature & Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
  N/A
  Also going to fix
  - #599

* **What is the new behavior (if this is a feature change)?**
Now the compass turtle can place blocks around the turtle with specific facing, which makes 3D printer easier.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their scripts due to this PR?)
No

* **Other information**:

- https://github.com/IntelligenceModding/Advanced-Peripherals-Documentation/pull/84

### CODE REVIEW

**Comment by SirEndii** [comment]
I am not 100% if I like this. This adds world manipulation functions to the Compass Turtle, whereas the Meta Turtles should actually be used for this
I know that the meta turtles currently don't have a place function
Would it be a better idea to add the accurate place feature to the meta turtles IF the compass is attached? The main problem I see with that, that it would use one peripheral slot for these two added functions

The implementation of these features looks good, I don't see anything in it, I would change

**Comment by zyxkad** [comment]
> Would it be a better idea to add the accurate place feature to the meta turtles IF the compass is attached?

Good point, I'll try that

**Comment by zyxkad** [comment]
However, can you review #597 first then? because merge two branch that changed same thing will be horrible

**Comment by zyxkad** [comment]
> Would it be a better idea to add the accurate place feature to the meta turtles IF the compass is attached? The main problem I see with that, that it would use one peripheral slot for these two added functions

Done

**Comment by SirEndii** [comment]
I see some commits here
Is this ready for review or a draft?

**Comment by zyxkad** [comment]
Yeah I just added some comment, it's ready for review

**Comment by zyxkad** [comment]
well I just noticed that `forward` is a verb, I should use `front` instead

**Comment by zyxkad** [comment]
Ready for review again

File: src/main/java/de/srendi/advancedperipherals/common/configuration/UnsafeConfig.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import net.minecraftforge.common.ForgeConfigSpec;
+
+public class UnsafeConfig {
+
+    private static ForgeConfigSpec.BooleanValue enableUnsafe;
+    private static ForgeConfigSpec.BooleanValue ignoreTurtlePeripheralItemNBT;
+
+    public static void build(final ForgeConfigSpec.Builder builder) {
+        enableUnsafe = builder.comment("By setting this value to true, I understand all operations below are danger to my adventure, and if they caused unexpected behaviour in my world, I will not consider it as AP's liability").define("enableUnsafe", false);
+        ignoreTurtlePeripheralItemNBT = builder.comment("Ignore turtle peripheral item's NBT when equipping. **YOU WILL LOST ALL NBT ON THE ITEM**").define("ignoreTurtlePeripheralItemNBT", false);
```
Comment (SirEndii) [inline src/main/java/de/srendi/advancedperipherals/common/configuration/UnsafeConfig.java:12]: Typos. lost -> lose, behaviour -> behavior (American English)

**Review by SirEndii — APPROVED** [review]

File: src/main/java/de/srendi/advancedperipherals/common/configuration/UnsafeConfig.java (hunk 1 above)
Comment (SirEndii) [reply src/main/java/de/srendi/advancedperipherals/common/configuration/UnsafeConfig.java:12]: I requested this change, forgot that I requested it, then I merged the PR 😮‍💨

Comment (zyxkad) [reply src/main/java/de/srendi/advancedperipherals/common/configuration/UnsafeConfig.java:12]: lol

**Pull request merged**